UI/CardScrollUI.gd

```diff
@@ -0,0 +1,71 @@
+extends Control
+
+var card_scene: PackedScene = preload("res://Cards/Card.tscn")
+
+var discard_pile: Array[CardBase] = CardManager.card_container.discard_pile
+var draw_pile: Array[CardBase]    = CardManager.card_container.draw_pile
+var deck: Array[CardBase]         = CardManager.card_container.default_deck
+var cards_to_display: Array[CardBase] = []
+var card_worlds: Array[CardWorld] = []
+var cardworld: CardWorld = null
+
+func _on_button_pressed() -> void:
+	queue_free()
+
+#Pressing escape will return you to the game
+func _input(event) -> void:
+	if event.is_action("escape"):
+		queue_free()
+
+
+
+func populate(parent_name: String) -> void:
```

Comment:
Let's have a unit test for this function. We're using this function for DiscardPile, DrawPile, and Deck, so having a unit test on the logic there would help. Checking that the text is correct when set, and that we're setting the proper Array as well.